File: tim-sort.java
import java.util.Arrays;
import java.util.Random;

public class TimSortExample {

    public static void main(String[] args) {
        // Create an array with 1 million Integer elements
        Integer[] array = new Integer[1_000_000];

        // Fill the array with random values
        Random random = new Random();
        for (int i = 0; i < array.length; i++) {
            array[i] = random.nextInt();
        }

        // Record the start time
        long startTime = System.currentTimeMillis();

        // Sort the array using TimSort (implicitly used by Arrays.sort for objects)
        Arrays.sort(array);

        // Record the end time
        long endTime = System.currentTimeMillis();

        // Calculate and print the sorting time
        System.out.println("Sorting 1 million integers took " + (endTime - startTime) + " milliseconds.");
    }
}

File: tim-sort.py
import random
import time

def main():
    # Create a list with 1 million random integers
    array = [random.randint(0, 1000000) for _ in range(1000000)]

    # Record the start time
    start_time = time.time()

    # Sort the list using TimSort (implicitly used by sort())
    array.sort()

    # Alternatively, you could use sorted() to achieve the same effect:
    # sorted_array = sorted(array)

    # Record the end time
    end_time = time.time()

    # Calculate and print the sorting time
    print(f"Sorting 1 million integers took {end_time - start_time:.2f} seconds.")

if __name__ == "__main__":
    main()
